bin/fit_posparams

```diff
@@ -0,0 +1,383 @@
+            if mode == 'static':
+                these_nominals = nominals
+            else:
+                these_nominals = nominals[posid]
+            argsdict = xytp_data
+            argsdict.update({'mode':mode,
+                             'nominals':these_nominals,
+                             'keep_fixed':params_to_keep_fixed,
+                             'description':job_desc})
+            if profile_one_job:
+                prof = cProfile.Profile()
+                params, fit_err, _ = prof.runcall(fitter.fit_params, **argsdict)
+                print(f'Results:\n{params}\nfit error = {fit_err}')
+                statsfile = 'stats'
+                prof.dump_stats(statsfile)
+                p = pstats.Stats(statsfile)
+                for sorter in ['cumtime', 'tottime']:
+                    p.strip_dirs()
+                    p.sort_stats(sorter)
+                    p.print_stats(15)
+                os.remove(statsfile)
+                return {posid:params}, fit_err
+            mp_results[case] = pool.apply_async(fitter.fit_params, kwds=argsdict)
+        
+        # retrieve results as they come in
+        while mp_results:
+            completed = set()
+            for case, result in mp_results.items():
+                if result.ready():
+                    completed.add(case)
+                    params, fit_err, job_desc = result.get()
+                    posid = case[case_enum['posid']]
+                    period_start_idx = case[case_enum['start_idx']]
+                    period_final_idx = case[case_enum['final_idx']]
+                    output['ANALYSIS_DATE'].append(now_str(date_format_data))
+                    output['POS_ID'].append(posid)
+                    output['DATA_START_DATE'].append(frames[posid]['DATE'][period_start_idx])
+                    output['DATA_END_DATE'].append(frames[posid]['DATE'][period_final_idx])
+                    output['NUM_POINTS'].append(period_final_idx - period_start_idx + 1)
+                    output['FIT_ERROR'].append(fit_err)
+                    for key in params:
+                        output[key].append(params[key])
+                    job_desc += '\n'
+                    job_desc += f'{posid}: best params = {dict_str(params)}\n'
+                    job_desc += f'{posid}: fit error = {fit_err:.3f}'
+                    printlog(f'job completed:\n{job_desc}\n')
+            for case in completed:
+                del mp_results[case]
+            time.sleep(0.05)
+    output_frame = pandas.DataFrame(output)
+    output_frame = output_frame.sort_values(['POS_ID','DATA_END_DATE'], ascending=True, ignore_index=True)
+    output_name = f'{timestamp_str}_results_{mode}.csv'
+    output_path = os.path.join(savedir, output_name)
+    output_frame.to_csv(output_path, index=False, date_format=date_format_data)
+    return output_frame, output_path
+        
+
+if __name__ == '__main__':
+    
+    # define analysis cases spanning the data history
```

Comment:
Here is where you would add `argparse` option parsing; see other scripts in desimeter/bin for examples.  This has the added benefit of making it safe to run `fit_posparams --help` and get help about input options without having it actually try to do anything.

Reply:
Ok. The purpose of the __main__ check was to ensure multiprocessing safety when running in certain environments (ipython, spyder). I don't see you using this in your scripts like desi_fvc_proc. Do you care about its presence?

Reply:
Including the `if __name__ == '__main__'` wrapper is good practice.  We haven't been strict about this elsewhere because a) we already have a strict separation of what gets imported (stuff in py/desimeter) vs. what gets run from the command line (stuff in bin/) and b) we haven't had people using spyder or other environments where not having it matters.